### DISCUSSION TITLE
Create Completions for Fig/Cloud Whisperer CLI

### DISCUSSION THREAD

**Discussion by miguelmig** [jdx/mise#3710] [Ideas]
Previously (and in documentation, still) [Fig](https://fig.io/) is now named [Amazon CodeWhisperer CLI](https://aws.amazon.com/blogs/devops/introducing-amazon-codewhisperer-for-command-line/).
It's an interactive tool for CLI that contains auto completions for a wide range of tools, and I find it very helpful. 
I noticed that there were no completions for `mise` available and I think they would be very helpful.
Considering that mise uses `clap` for CLI and Fig already has a tutorial for [this](https://fig.io/docs/guides/integrating/integrations/clap) I'd say it's feasible to add completion support

**Comment by miguelmig** [comment]
Considering the comment on previous closed PR, created the following PR for usage: https://github.com/jdx/usage/pull/148. Assuming it gets approved and merged, the next steps is to add a task on mise repo for generating the fig spec and auto-publishing to https://github.com/withfig/autocomplete

**Comment by jdx** [comment]
this is done right @miguelmig ?

**Comment by miguelmig** [comment]
Not quite, it seems that the linter action on the autocomplete PR of withfig doesnt like the way we auto generate the spec, theres work to be done

**Comment by jdx** [comment]
@miguelmig I had a look myself but didn't get too far, I'll see if I can pick it back up again soon.

**Comment by jeremiah-snee-openx** [comment]
Semi Related Conversation but focused on current state and requirements?
- https://github.com/jdx/mise/discussions/4852
